File: ruby/raindrops/raindrops.rb
class Raindrops
  def convert(a)
    e = []
    e << "Pling" if prime?(a,3)
    e << "Plang" if prime?(a,5)
    e << "Plong" if prime?(a,7)

    e.empty? ? a.to_s : e.join
  end

  private

  def prime?(a,n)
    a % n == 0
  end
end

File: haskell/raindrops/Raindrops.hs
module Raindrops (convert) where

convert :: Int -> String
convert n = if (not . null) factors then showFactors else show n
  where
    factors      = filter isFactor [3,5,7]
    isFactor     = (== 0) . mod n
    showFactors  = concatMap showFactor factors
    showFactor 3 = "Pling"
    showFactor 5 = "Plang"
    showFactor _ = "Plong"
